[Clevon Wong Jun Yong] iP

clevon-w opened the pull request:
### DukePro
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

-text-based
-easy to learn
-~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

clevon-w committed:
echo functionality

clevon-w committed:
added javadoc and cleaned printing

clevon-w committed:
cleaning up main function

clevon-w committed:
abstracted user req parsing

clevon-w committed:
added functionality with OOP standards

clevon-w committed:
rename function for clarity

clevon-w committed:
added javadoc

clevon-w committed:
added parent exception for future proofing

clevon-w committed:
functionality for mark and unmark

clevon-w committed:
added normalisation of whitespace function to user inputs

clevon-w committed:
added javadoc for level3 functionality

clevon-w committed:
added 3 subclasses of Task

clevon-w committed:
added end time to Event

clevon-w committed:
initial functionality for parsing and response

clevon-w committed:
added count to print response

clevon-w committed:
customised error message

clevon-w committed:
fix minor bugs in response and parser

clevon-w committed:
resolved bugs for mark and unmark edge cases

clevon-w committed:
updated error message for missing arg

clevon-w committed:
added test cases into input.txt and expected results into EXPECTED.TXT

clevon-w committed:
didnt know level5 is exceptions, paraphrased error messages to fit assignment.

clevon-w committed:
bugs for testing and errror messages

clevon-w committed:
added delete functionality

clevon-w committed:
added auto testing for lvl 6

clevon-w committed:
created basic local storage class with save and load functionality (untested)

clevon-w committed:
Update functionality to support ToDoList and not just ArrayList<String>

clevon-w committed:
Add save and load to main function

clevon-w committed:
Add LocalDate support with error handling

clevon-w committed:
Add throws to function definitions

clevon-w committed:
Merge branch 'branch-Level-7'

clevon-w committed:
Merge branch 'branch-Level-8'

clevon-w committed:
Update todolist naming to tasklist

clevon-w committed:
Change LocalStorage to Storage

clevon-w committed:
Update InputParser with Parser

clevon-w committed:
Add Ui class

clevon-w committed:
Merge branch 'add-gradle-support'

clevon-w committed:
Configure build.gradle, add checkstyle

clevon-w committed:
Debug gradle checkstyle

clevon-w committed:
Test gradle checkstyles. Rectify style errors.

clevon-w committed:
Rectify style errors in MissingArgumentException

clevon-w committed:
Add jUnit. Test jUnit. Found bugs. Debugged bugs.

clevon-w committed:
Check main and test complies with coding standards

clevon-w committed:
Revert "Check main and test complies with coding standards"

This reverts commit 15e5641044d6e9c03b972d5eeae72c64bd09bf3a.

clevon-w committed:
Add missing JavaDoc for all functions

clevon-w committed:
Check main and test complies with coding standards

clevon-w committed:
Add find functionality to task list

clevon-w committed:
Add JUnit for FindResponse

clevon-w committed:
Ran Checkstyle for FindResponseTest

clevon-w committed:
Merge branch 'branch-A-JavaDoc'

clevon-w committed:
Merge branch 'branch-A-CodingStandard'

clevon-w committed:
Merge branch 'branch-Level-9'

dsja612 commented on data/duke.txt:
```diff
@@ -1,3 +1,4 @@
 T | 1 | test
 D | 0 | test | 2023-01-27
 T | 0 | test
+T | 0 | something
```
Perhaps you could add your private `duke.txt` local file to your `.gitignore` file? Not sure if it's a good idea/practice to commit local data files

Nice formatting of tasks

dsja612 commented on src/main/java/response/DeadlineResponse.java:
```diff
@@ -0,0 +1,84 @@
+        this.deadline = deadline;
+    }
+
+    /**
+     * Creates a new deadline in the to do list
+     * @param taskList The to do list to create a new deadline in
+     * @return String to indicate that a new deadline was created successfully
+     */
```
Should also describe the exceptions thrown in this method in JavaDocs header

i.e. `@throws fooException If ....`

Most of your method headers have that except for a few. Good job!

dsja612 reviewed (commented):
Looks great overall.
Great job!

Mehvin commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,136 @@
+        int l = s.length();
+        StringBuilder sb = new StringBuilder(l);
+
+        boolean lastWasWhite = false;
+        boolean modified = false;
```
Perhaps better naming can be used for boolean variables?
```suggestion
        boolean wasLastWhite = false;
        boolean isModified = false;
```

Mehvin commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,36 @@
+     * Prints formatted response to the console.
+     * @param s string that will be printed
+     */
+    public void printResponse(String s) {
+        String p = String.format("\t____________________________________________________________\n"
+                + "\t %s\n"
+                + "\t____________________________________________________________\n", s);
+        System.out.println(p);
```
Great job on the formatting here 👍

Mehvin commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,136 @@
+            return new FindResponse(this.inputContent);
+        default:
+            throw new MissingArgumentException(
+                    "I'm sorry, but I don't know what that means :-("
+                            + "\n\t  To create a new todo, use 'todo ...',"
+                            + "\n\t  To create a new event, use 'event ... /from ... /to ...',"
+                            + "\n\t  To create a new deadline, use 'deadline ... /by ...',"
+                            + "\n\t  To list all tasks, use 'list',"
+                            + "\n\t  To mark a task, use 'mark ' with a number,"
+                            + "\n\t  To unmark a task, use 'unmark ' with a number,"
+                            + "\n\t  Finally to exit the program, use 'bye'!");
```
I like that you gave all the possible commands format the user should follow!

Mehvin reviewed (commented):
Overall great code quality, I also like that all JavaDocs are in placed! 👍

clevon-w committed:
Update build.gradle to set up Java FX

clevon-w committed:
Lean JavaFX basics and create a basic GUi for Duke

clevon-w committed:
Add functionality for getting Duke response, remove echo

clevon-w committed:
Add exit functionality for GUi

clevon-w committed:
Add welcome message to GUi

clevon-w committed:
Add welcome message functionality

clevon-w committed:
Edit string response to better fit GUi

clevon-w committed:
Add padding for text

clevon-w committed:
Add varargs

clevon-w committed:
Merge conflicts from Varargs

clevon-w committed:
The current implemention does not use any assertions.

Assertions should be implemented because they can be a useful tool for improving the quality, clarity, and maintainability of the code.

Some assertions are being added to Storage.java and Duke.java.

It is done this way because without them, there is nothing checking if the inputs are missing or not which could potentially lead to run time errors.

clevon-w committed:
Code quality: shorten Storage::loadTaskList

Storage::loadTaskList is well over 30 lines

Long and verbose methods make it unreadable and difficult to debug

Extracting out code chunks into separate methods makes the method shorter and easier to debug.
For example, when a bug occurs from Storage::loadTaskList, one has to go through the whole piece of code to debug but if it was separated, the debugger will be able to understand at a glance what the loadTaskList method does and zoom into the bug

clevon-w committed:
Code Quality: Shorten EventResponse::exec

EventResponse::exec is well over 30 lines

The chunk of code that is responsible for parsing and checking the inputs for errors are extracted out into another method

Long and verbose methods are very unreadable and difficult to debug.
For example, if the method is really long, a bug can come from either the parsing of inputs or the creation of a new Event. However, now that it is separated, at a glance, one can tell where the bug is coming from.

clevon-w committed:
Merge pull request #1 from clevon-w/branch-A-Assertions

The current implemention does not use any assertions.

clevon-w committed:
Resolve merge conflicts between master and branch-A-CodeQuality

clevon-w committed:
Merge pull request #2 from clevon-w/branch-A-CodeQuality

Critical review of code quality

clevon-w committed:
Add gradle.yaml file for github actions

clevon-w committed:
Add sort into parse()

clevon-w committed:
Update capabilities string with sort

clevon-w committed:
Implement compareTo for Tasks

clevon-w committed:
Add SortResponse class to exec sort command

clevon-w committed:
Fix checkStyles

clevon-w committed:
Fix more checkStyles and remove printing of input to console

clevon-w committed:
Debug loading event type from data

clevon-w committed:
Fix checkStyle in MainWindow

clevon-w committed:
Fix gradle test files

clevon-w committed:
Remove unused IO tests

clevon-w committed:
Merge pull request #4 from clevon-w/branch-C-sort

C-sort

clevon-w committed:
Add screenshot

clevon-w committed:
Update gitignore

clevon-w committed:
removing data

clevon-w committed:
Add User Guide

clevon-w committed:
Change mainclassName in build.gradle

clevon-w committed:
edit read me
